src/python/pants/build_graph/address_test.py

```diff
@@ -75,6 +75,10 @@ def assert_parsed(
     assert_parsed(
         "a#gen@k1=v1", generated_component="gen", path_component="a", parameters={"k1": "v1"}
     )
+    assert_parsed("a@t", path_component="a@t")
+    assert_parsed("a@=", path_component="a@=")
```

Comment:
This one is vaguely questionable: but as it stands, we don't disallow `=` in pathnames (although that would _probably_ be safe...?)

Reply:
I think it is probably safer to allow it. I do worry that using some of the symbols like @ Will make addresses even harder to understand. But that is probably less offensive than failing to build and forcing users to rename their file?

Reply:
yea, forcing to rename files is a no-go imo, there ought to be escape hatches to navigate, even if ugly rather than going into a ditch.